### DISCUSSION TITLE
Automatic header?

### DISCUSSION THREAD

**Discussion by jtppragg** [marp-team/marp#212] [Q&A] [answered]
Is there an easy way to use the last HTML heading as the current slide header? I would find this very helpful in those cases where the same section spans several slides. Currently I manually set _header for those slides, but this leads to redundancies.

Thank you in advance!

**Comment by yhatt** [comment, marked as answer]
Try using `header` instead of `_header`. You can change the header for subsequent slides at once.

```markdown
<!-- header: Heading -->
# Heading

page 1

---

page 2

---

page 3

---

<!-- header: Next section -->
# Next section

page 4

---

page 5

---

page 6
```

In this example, the header content "Heading" has in 1-3 slides, and "Next section" has in 4-6.

> You have to need to write a Marp plugin if don't want defining header directives `<!-- header: xxx -->`.

**Reply by jtppragg** [reply]
Thank you Yuki! I have already tried that, but there is still this redundancy:

```markdown
<!-- header: 'I have to write this twice for every section' -->
# I have to write this twice for every section
```
If anyone knows an existing plugin for this, please tell me.
